### ISSUE TITLE
Synced blocks not displaying in different pages

### ISSUE THREAD

**Issue by cfranc0** [NotionX/react-notion-x#183]
Hello!
I'm having some doubts about the synced block support.
In the [demo](https://react-notion-x-demo.transitivebullsh.it/sync-blocks-8bcd65801a5d450fb7218d8890a38c29) the blocks appear to work as intended and they do if the original block and the clones are in the same page.
But if I create the block in page A:
![image](https://user-images.githubusercontent.com/20073894/146672371-e1e13223-4f8e-4081-93b3-3aec7a40c3f7.png)
And have the synced block in page B this is what I can see:
![image](https://user-images.githubusercontent.com/20073894/146672377-07498d03-d703-46f9-ab35-03e14cef446c.png)

This seems to be the case also with file links, they are in the page and visible but no longer take you to their destination.

Please, do let me know if there is something wrong with my use of the library or if it's a known problem.
Have a great day!

**Comment by transitive-bullshit** [comment]
I agree that the issue is likely how we resolve which blocks to load up front. We support the 95% use case really well, but there are always blocks (and version of different block types) which aren't handled properly.

Notion's renderer uses a client-side API call to load additional blocks on-the-fly, which is definitely possible w/ `react-notion-x`, but it isn't handled out of the box just yet.

Ideally, if `NotionRenderer` encounters a block it doesn't have pre-loaded, it would make a call to the backend to load it like Notion does. This is also how we'd support larger collections.

If anyone wants to investigate, PRs are welcome.
